r1cs/gadgets/std/src/groups/nonnative/nonnative_group_gadget.rs

```diff
@@ -0,0 +1,1015 @@
+use algebra::{AffineCurve, BitIterator, Field, PrimeField, ProjectiveCurve, SWModelParameters, SquareRootField, curves::short_weierstrass_jacobian::{GroupAffine as SWAffine, GroupProjective as SWProjective}};
+
+use r1cs_core::{ConstraintSystem, SynthesisError};
+
+use crate::{Assignment, ToBitsGadget, ToBytesGadget, alloc::{AllocGadget, ConstantGadget}, boolean::Boolean, fields::{FieldGadget, nonnative::nonnative_field_gadget::NonNativeFieldGadget}, prelude::EqGadget, select::{CondSelectGadget, TwoBitLookupGadget}, uint8::UInt8};
+use std::{borrow::Borrow, marker::PhantomData};
+
+use super::NonNativeGroupGadget;
+
+#[derive(Derivative)]
+#[derivative(Debug, Clone)]
+pub struct GroupAffineNonNativeGadget<
+    P: SWModelParameters<BaseField = SimulationF>,
+    ConstraintF: PrimeField,
+    SimulationF: PrimeField + SquareRootField,
+    F: FieldGadget<SimulationF, ConstraintF>,
+> {
+    pub x:   NonNativeFieldGadget::<SimulationF, ConstraintF>,
+    pub y:   NonNativeFieldGadget::<SimulationF, ConstraintF>,
+    pub infinity:   Boolean,
+    _params: PhantomData<P>,
+    _engine: PhantomData<ConstraintF>,
+    _simF: PhantomData<SimulationF>,
+    _F: PhantomData<F>
+}
+
+
+impl<P, ConstraintF, SimulationF, F> NonNativeGroupGadget<SWProjective<P>, ConstraintF, SimulationF>
+for GroupAffineNonNativeGadget<P, ConstraintF, SimulationF, F>
+    where
+        P: SWModelParameters<BaseField = SimulationF>,
+        ConstraintF: PrimeField,
+        SimulationF: PrimeField + SquareRootField,
+        F: FieldGadget<SimulationF, ConstraintF>,
+{
+    type Value = SWProjective<P>;
+    type Variable = (F::Variable, F::Variable);
```

Comment:
No corresponding variable for a non native field element. This should be just empty ()

Reply:
Done